server/src/router/newsletter.js

```diff
@@ -1,10 +1,38 @@
-import express from 'express'
-const router=express.Router()
-import User from '../model/user.js'
-import {mailer} from '../controllers/newsletter.js'
-router.get('/',async (req,res)=>{
-    
-    res.send( await User.find({newsletterIsSubscribed:true}));
-})
-router.get('/email',mailer)
-export default router;
\ No newline at end of file
+import express from "express";
+const router = express.Router();
+import transporter from "../config/nodemailer.js";
+import User from "../model/user.js";
+import { mailer } from "../controllers/newsletter.js";
+
+router.get("/", async (req, res) => {
+  res.send(await User.find({ newsletterIsSubscribed: true }));
+});
+
+router.get("/email", mailer);
+
+router.post("/", async (req, res) => {
+  try {
+    const { subject, body } = req.body;
+
+    const SubscribedUsers = await User.find({ newsletterIsSubscribed: true });
+    if (!SubscribedUsers || SubscribedUsers.length == 0) {
+      return res.status(404).json({ message: "No subscribed user found" });
+    }
+
+    const recipients = SubscribedUsers.map((user) => user.email);
+
+    await transporter.sendMail({
```

Comment:
This method sends a single email with all user emails. which is a bad practice, as we are revealing our user email in public. Instead put them in bcc to hide emails

Reply:
Done